x-pack/filebeat/input/awss3/s3_objects.go

```diff
@@ -448,7 +452,10 @@ func (p *s3ObjectProcessor) createEvent(message string, offset int64) beat.Event
 			},
 		},
 	}
-	event.SetID(objectID(p.s3ObjHash, offset))
+	if offset >= 0 {
+		event.Fields.Put("log.offset", offset)
+		event.SetID(objectID(p.s3ObjHash, offset))
```

Comment:
Why do we also skip setting the event ID? Is there something that uses this for idempotency?

Reply:
That was the stimulus for the change; in the case that there is no way to know the offset, we would end of making an `@metadata._id` for the document that is shared for **all documents**  from the same object which would result in invalid document duplicates being handled by the index, and loss of data. By allowing a way to signal to the input that there is no way to differentiate between documents from the data source we allow downstream to know that it needs to fill the gap.